atlascope/core/job_types/average_color.py

```diff
@@ -17,19 +17,23 @@ def run(job_id: str, original_dataset_id: str):
     original_dataset = Dataset.objects.get(id=original_dataset_id)
     job = Job.objects.get(id=job_id)
 
-    input_image = Image.open(io.BytesIO(original_dataset.content.read())).convert("RGB")
-
-    average_color = [int(value) for value in np.average(input_image, axis=(0, 1))]
-    output_image = Image.new(mode="RGBA", size=(200, 200), color=tuple(average_color))
-
-    job.resulting_datasets.add(
-        save_output_dataset(
-            original_dataset,
-            job.investigation,
-            'Average Color',
-            output_image,
-            {'rgba': average_color},
+    try:
+        input_image = Image.open(io.BytesIO(original_dataset.content.read())).convert("RGB")
+
+        average_color = [int(value) for value in np.average(input_image, axis=(0, 1))]
+        output_image = Image.new(mode="RGBA", size=(200, 200), color=tuple(average_color))
+
+        job.resulting_datasets.add(
+            save_output_dataset(
+                original_dataset,
+                job.investigation,
+                'Average Color',
+                output_image,
+                {'rgba': average_color},
+            )
         )
-    )
-    job.complete = True
-    job.save()
+        job.complete = True
+        job.save()
+    except Exception as e:
+        job.failure = str(e)
+        job.save()
```

Comment:
I *think* you could factor out this `job.save()` and the one from line 36 into a `finally` clause for this try block. It is not urgent that you do that, but it would be a nice use of `finally` for the "cleanup" action of saving the Job model.

Reply:
Changed in [91b8359](https://github.com/atlascope/atlascope/pull/148/commits/91b835953f6a353adee000355d873ff367033a6f)